decoder.go

```diff
@@ -258,6 +227,52 @@ func DecodeTime(bz []byte) (t time.Time, n int, err error) {
 	return
 }
 
+func decodeSeconds(bz *[]byte) (int64, int, error) {
+	// Optionally decode field number 1 and Typ3 (8Byte).
+	// only slide if we need to:
+	n := 0
+	fieldNum, typ, _n, err := decodeFieldNumberAndTyp3(*bz)
+	if err != nil {
+		return 0, n, err
+	}
+	if fieldNum == 1 && typ == Typ3_8Byte {
+		slide(bz, &n, _n)
+		_n = 0
+		sec, _n, err := DecodeInt64(*bz)
+		if slide(bz, &n, _n) && err != nil {
+			return 0, n, err
+		} else {
+			return sec, n, err
+		}
+	} else if fieldNum == 2 && typ == Typ3_4Byte {
+		// skip: do not slide, no error, will read again
+		return 0, n, nil
+	} else {
+		return 0, n, fmt.Errorf("expected field number 1 <8Bytes> or field number 2 <4Bytes> , got %v", fieldNum)
```

Comment:
Is there a general error strategy for amino? While totally out of scope of this PR, errors should be properly constructed (sentinels, or types) to give control to the caller. Esp. important with low-level libs.

Reply:
Agreed! But out of scope for this PR as you said.